### ISSUE TITLE
In caffe-windows, compute_image_mean keeps increasing to use out the memory.

### ISSUE THREAD

**Issue by XiongDuan** [BVLC/caffe#4329]
I've compiled and run the Caffe-Windows successfully. But the memory usage of compute_image_mean keeps on increasing and finally use out all the memory ,and makes the windows systeam very slow. 

In my project, I have 1.5 millions images, however the memory will be used out when 0.1 million files being processed.

I searched on the Internet but cannot find any similar issues like me.  I thought it is an issue of memory leak, but I can't find out where the bug is. And if there is nobody having the same situation with me, maybe it's my computer systeam's problem.

**Comment by seanbell** [comment]
Why are you computing the image mean across 1.5 million images?  Use a random subsample.  Or even better, just use the imagenet mean if your images are "natural"-ish.

**Comment by XiongDuan** [comment]
You're right. Thanks a lot for your reply. But the problem is I found that not compute_image_mean but all the program compiled in this solution have a big memory consumption. I trained my model with 40000 iterations and with batch_size of 256, but it cost me two days to finish it. During that time, the computer is very slow. I cannot do anything else. 
I'm not good at programming, are there any problems in my computer ? or is just the bug of caffe-windows?

**Comment by seanbell** [comment]
If you have only one GPU and use it both to run the monitor you're using, and train a net, then yes the GUI will be very slow and unresponsive.  You'd have the same issue if you tried to render a 3D scene using Blender on the GPU, for example.  This is not a bug, bur rather the fact that caffe uses as much of the GPU as possible.

I'm going to close the issue since this is more about how to use caffe.  Please follow up on the mailing list if you have more questions.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

**Comment by aysegulucar2016** [comment]
How did you  compiled and run the Caffe-Windows?

**Comment by XiongDuan** [comment]
Just use the branch windows of caffe. All the steps are very easy, you do nothing but wait it to work.

**Comment by willyd** [comment]
Does this issue still exist with the latest windows branch @XiongDuan?
